pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -1203,21 +1203,36 @@ func (m *ManagerImpl) ShouldResetExtendedResourceCapacity() bool {
 }
 
 func (m *ManagerImpl) isContainerAlreadyRunning(logger klog.Logger, podUID, cntName string) bool {
-	cntID, err := m.containerMap.GetContainerID(podUID, cntName)
-	if err != nil {
-		logger.Error(err, "Container not found in the initial map, assumed NOT running", "podUID", podUID, "containerName", cntName)
+	// Check if ANY container for this pod/container name is running.
+	// This handles the case where a container restarted before kubelet restart,
+	// so the containerMap might have multiple entries (old exited + new running).
+	// We need to check all of them to see if any are running.
```

Comment:
Is there any scenario where the old (exited) container ID might point to a container that for some reason should remain associated (e.g. for cleanup)? In other words: does the new logic risk skipping legitimate containers inadvertently? It might be worthwhile to consider edge cases like multiple containers, init containers, or other special container states.